src/lib.rs

```diff
@@ -386,47 +385,36 @@ mod tests {
         ),
         (
           "file:///a/test02.ts",
-          Ok(("file:///a/test02.ts", None, r#"export const b = "b";"#)),
+          Ok((
+            "file:///a/test02.ts",
+            None,
+            r#"export const b = "b"; let t;"#,
+          )),
         ),
       ],
       vec![],
     );
     let root_specifier =
       ModuleSpecifier::parse("file:///a/test01.ts").expect("bad url");
+    let test02_specifier =
+      ModuleSpecifier::parse("file:///a/test02.ts").expect("bad url");
     let mut parser = crate::ast::DefaultAstParser::new();
-    let graph = create_graph(
+    create_graph(
       root_specifier.clone(),
-      loader,
+      &mut loader,
       None,
       None,
       Some(&mut parser),
     )
     .await;
-    assert_eq!(graph.modules.len(), 2);
-    assert_eq!(graph.root, root_specifier);
-    let maybe_root_module = graph.modules.get(&root_specifier);
-    assert!(maybe_root_module.is_some());
-    let root_module_slot = maybe_root_module.unwrap();
-    if let ModuleSlot::Module(module) = root_module_slot {
-      assert_eq!(module.dependencies.len(), 1);
-      let maybe_dependency = module.dependencies.get("./test02.ts");
-      assert!(maybe_dependency.is_some());
-      let dependency_specifier =
-        ModuleSpecifier::parse("file:///a/test02.ts").unwrap();
-      let dependency = maybe_dependency.unwrap();
-      assert!(!dependency.is_dynamic);
-      if let Resolved::Specifier(resolved_specifier, _) = &dependency.maybe_code
-      {
-        assert_eq!(resolved_specifier, &dependency_specifier);
-      } else {
-        panic!("unexpected resolved slot");
-      }
-      assert_eq!(dependency.maybe_type, Resolved::None);
-      let maybe_dep_module_slot = graph.get(&dependency_specifier);
-      assert!(maybe_dep_module_slot.is_some());
-    } else {
-      panic!("unspected module slot");
-    }
+    let root_ast = parser.get_ast(&root_specifier).unwrap();
```

Comment:
This fundamentally changes what is being tested here... It might be nice to ensure the parser gets populated with the AST, but this test was to help ensure the public API surface of the graph behaved as expected, not that the ast was populated.  We should add a test, not delete one, IMO.

Reply:
> We should add a test, not delete one, IMO.

I think you might be viewing the diff between two commits because it's actually adding a new test. I initially just copied the test above and I didn't think it was worth retesting all the other stuff so I just did a very basic test that the ASTs get populated.